src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -23,13 +23,20 @@ The .NET Foundation licenses this file to you under the MIT license.
     <IlcBuildTasksPath Condition="'$(IlcBuildTasksPath)' == ''">$(MSBuildThisFileDirectory)..\tools\netstandard\ILCompiler.Build.Tasks.dll</IlcBuildTasksPath>
     <TargetOS Condition="$(RuntimeIdentifier.StartsWith('win'))">windows</TargetOS>
     <TargetOS Condition="$(RuntimeIdentifier.StartsWith('osx'))">osx</TargetOS>
+    <TargetOS Condition="$(RuntimeIdentifier.StartsWith('maccatalyst'))">maccatalyst</TargetOS>
```

Comment:
Should we use here the same principle as above with: `<OSIdentifier Condition="'$(OSIdentifier)' == '' and $(RuntimeIdentifier.StartsWith('win'))">win</OSIdentifier>` in the "first-match-wins" approach, and include `'$(TargetOS)' == ''` condition in this case as well?

Reply:
Correct, otherwise it won't select the first match.

Reply:
Up until now, these are not meant to be set from the outside, so we should also remove it from `OSIdentifier` since there is no use-case for it. We try to limit unnecessary extension points which become hard to maintain later and to justify during refactoring. I am planning to look into deduplicating these two lists as we really need a single list of hardcoded platforms for shipping targets (ideally we shouldn't be even needing that and directly rely on SDK provided properties...).

Reply:
I understand your concern and agree that if these properties are not meant to be extensible, the conditions should be removed (in both places). 
However, if this is going to be changed in the near future in any case, can we agree on the implementation for this PR, so we can close it?
@jkotas what do you think?

Reply:
>  can we agree on the implementation for this PR, so we can close it?

Sure, I'll follow-up.

Reply:
> @jkotas what do you think?

I agree that this can use refactoring that can be done as a follow up (not required for this PR).